apis/v1alpha2/validation/httproute.go

```diff
@@ -166,3 +144,34 @@ func validateHTTPPathMatch(path *gatewayv1a2.HTTPPathMatch, fldPath *field.Path)
 	}
 	return allErrs
 }
+
+// validateHTTPRouteFilterTypeMatchesValue validates that each filter type matches
+//the filter value and that values are not empty.
+func validateHTTPRouteFilterTypeMatchesValue(filter gatewayv1a2.HTTPRouteFilter, path *field.Path) field.ErrorList {
+	var errs field.ErrorList
+	switch filter.Type {
+	case gatewayv1a2.HTTPRouteFilterExtensionRef:
+		if filter.ExtensionRef == nil || (gatewayv1a2.LocalObjectReference{}) == *filter.ExtensionRef {
```

Comment:
```suggestion
		if filter.ExtensionRef == nil || gatewayv1a2.LocalObjectReference{} == *filter.ExtensionRef {
```

I think that this is the correct pattern to apply here. We don't need deep equal, because if the filter is empty, the top-level fields will all equal the top level fields of the empty struct.

Reply:
I've changed these and others. However, in the case of `HTTPRequestHeaderFilter` it cannot be compared as it contains slices. Here is the error : `struct containing []ref cannot be compared`. So what I did instead was to check that those slices are not nil, instead of using deep equal.

Reply:
OK, that sounds ok :+1: